src/librustc/hir/map/definitions.rs

```diff
@@ -481,6 +481,11 @@ impl Definitions {
         self.node_to_hir_id[node_id]
     }
 
+    #[inline]
+    pub fn def_index_to_node_id(&self, def_index: DefIndex) -> ast::NodeId {
+        self.as_local_node_id(DefId::local(def_index)).unwrap()
```

Comment:
You could copy the `Some` part of `as_local_node_id` here and call this function from there instead of unwrapping here